sktime/forecasting/boxcox_bias_adjusted_forecaster.py

```diff
@@ -0,0 +1,188 @@
+        -------
+        self : returns an instance of self.
+        """
+        self.boxcox_transformer_ = BoxCoxTransformer(lmbda=self.lmbda)
+        y_transformed = self.boxcox_transformer_.fit_transform(y)
+
+        self.forecaster_ = self.forecaster.clone()
+        self.forecaster_.fit(y_transformed, X, fh)
+
+        # Check if the wrapped forecaster supports variance prediction
+        if not hasattr(self.forecaster_, "predict_var"):
+            raise ValueError(
+                "The wrapped forecaster must implement a `predict_var` method "
+                "to enable bias adjustment."
+            )
+
+        return self
+
+    def _predict(self, fh, X=None):
+        """Forecast time series at future horizon.
+
+        Parameters
+        ----------
+        fh : ForecastingHorizon
+            The forecasting horizon with the steps ahead to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables.
+
+        Returns
+        -------
+        y_pred : pd.DataFrame
+            Bias-adjusted point predictions.
+        """
+        y_pred_transformed = self.forecaster_.predict(fh, X)
+        variance = self.forecaster_.predict_var(fh, X)
+
+        y_pred = self.boxcox_transformer_.inverse_transform(y_pred_transformed)
+        y_adjusted = self._apply_bias_adjustment(y_pred, variance)
+
+        return y_adjusted
+
+    def _predict_interval(self, fh, X=None, coverage=None):
+        """Compute prediction intervals for the forecasts.
+
+        Parameters
+        ----------
+        fh : ForecastingHorizon
+            The forecasting horizon with the steps ahead to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables.
+        coverage : list of float, optional (default=None)
+            Confidence levels for prediction intervals.
+
+        Returns
+        -------
+        pred_int : pd.DataFrame
+            Prediction intervals.
+        """
+        pred_int_transformed = self.forecaster_.predict_interval(fh, X, coverage)
+        pred_int = self.boxcox_transformer_.inverse_transform(pred_int_transformed)
+
+        variance = self.forecaster_.predict_var(fh, X)
+
+        lower_adjusted = self._apply_bias_adjustment(pred_int["lower"], variance)
+        upper_adjusted = self._apply_bias_adjustment(pred_int["upper"], variance)
+
+        return pd.concat([lower_adjusted, upper_adjusted], axis=1)
+
+    def _apply_bias_adjustment(self, y, variance):
+        """Apply bias adjustment using Hyndman's method.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Predictions in the original scale.
+        variance : pd.DataFrame
+            Variance of predictions in the transformed scale.
+
+        Returns
+        -------
+        y_adjusted : pd.DataFrame
+            Bias-adjusted predictions.
+        """
+        lmbda = self.boxcox_transformer_.lmbda_
+        epsilon = 1e-6
+
+        if abs(lmbda) < epsilon:
```

Comment:
this if/else kinda works, but it looks like a hack. I would suggest trying Taylor expansion around lambda = zero. I do not have the exact terms, but this should render the limit numerically stable if the right terms are used.